File: Code/StealthBatteryCharger/Feather/BatterySynoptics/Box.cpp
/*****************************************************************************
 **
 **  Project:                      Stealth Battery Charger
 **
 **                             Copyright (c) 2023
 **                                James A Wright
 **                            All Rights Reserved
 **
 **  File Description:
 **      This displays A Box
 **
 *****************************************************************************/

//--------------------------------------------------------------------
//  Included Files
//--------------------------------------------------------------------
#include "Box.h"

/*****************************************************************************
 **
 **  Purpose:
 **      Constructor
 **
 **  Notes: none
 **
 *****************************************************************************/
Box::Box(int16_t oneX, int16_t oneY, int16_t twoX, int16_t twoY, int16_t size, Color color) : DrawingObjects(),
  mOne(oneX, oneY, twoX, oneY, size, color),
  mTwo(twoX, oneY, twoX, twoY, size, color),
  mThree(twoX, twoY, oneX, twoY, size, color),
  mFour(oneX, oneY, oneX, twoY, size, color)
{
    mType = DrawingObjects::TYPE_BOX;
}

/*****************************************************************************
 **
 **  Purpose:
 **      Seet the colors of the lines
 **
 **  Notes: 
 **      None
 **
 *****************************************************************************/
void Box::SetColor(Color color)
{
  mOne.SetColor(color);
  mTwo.SetColor(color);
  mThree.SetColor(color);
  mFour.SetColor(color);
}

/*****************************************************************************
 **
 **  Purpose:
 **      Draw a Line in the saved color
 **
 **  Notes: 
 **      None
 **
 *****************************************************************************/
void Box::Draw()
{
  mOne.Draw();
  mTwo.Draw();
  mThree.Draw();
  mFour.Draw();
}

File: Code/StealthBatteryCharger/Windows/C#/DragAndDropTest/DragAndDropTest/Box.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Windows.Forms;
using System.Drawing;

namespace DragAndDropTest
{
    class Box : DrawingObjects 
    {
        Line mOne;
        Line mTwo;
        Line mThree;
        Line mFour;

        public Box(double x1, double y1, double x2, double y2, double size, Color color) : base()
        {
            mOne   = new Line(x1, y1, x2, y1, size, color);
            mTwo   = new Line(x2, y1, x2, y2, size, color);
            mThree = new Line(x2, y2, x1, y2, size, color);
            mFour  = new Line(x1, y1, x1, y2, size, color);
        }

        public override void SetColor(Color color)
        {
            mOne.SetColor(color);
            mTwo.SetColor(color);
            mThree.SetColor(color);
            mFour.SetColor(color);
        }

        public override void Draw(Graphics graph, double xOffset, double yOffset)
        {
            mOne.Draw(graph, xOffset, yOffset);
            mTwo.Draw(graph, xOffset, yOffset);
            mThree.Draw(graph, xOffset, yOffset);
            mFour.Draw(graph, xOffset, yOffset);
        }


    }
}
